### PR TITLE
Generate: add missing `**model_kwargs` in sample tests

### PR DESCRIPTION
**Pull request by gante**
# What does this PR do?

One call of the `sample`-related tests was missing `**model_kwargs`, which MAY explain the random failures we were seeing. 

I've run all `test_sample_generate_dict_output` tests 100x with no failures. Before this change, it was failing once every ~10 calls of `py.test tests/ -k test_sample_generate_dict_output`.

### CODE REVIEW

File: tests/generation/test_generation_utils.py [hunk 1]
```diff
@@ -327,6 +327,7 @@ def _sample_generate(
             remove_invalid_values=True,
             **logits_warper_kwargs,
             **process_kwargs,
+            **model_kwargs,
```
Comment (gante) [inline tests/generation/test_generation_utils.py:330]: The dict only carries the attention mask :)

**Review by ydshieh — APPROVED** [review]
Thank you @gante . We can finally sleep better without seeing this pop again :-)

It is somehow mysterious that without passing attn mask, we have more frequent test failure.
It might be a good idea to keep this on a TODO list somewhere, either a comment above the changed line, or on Notion.

**Comment by gante** [comment]
@ydshieh when the attention mask is not passed, it is inferred from the input ([here](https://github.com/huggingface/transformers/blob/e54a1b49aa6268c484625c6374f952f318914743/src/transformers/generation_utils.py#L490)).

Depending on the pad and eos tokens, the mask inferred from the random input tokens may have issues. This automatic attention mask is a big source of issues in general, especially for the tests :( 

Passing the mask explicitly is always preferred!
